### PR TITLE
Bug fix: locking when going to background

### PR DESCRIPTION
**Pull request by myxmaster**
# Description

The commit 22a9c8c broke the setting "Require login after app returns from background" because it did not logout when the user was logged in. Not sure if this breaks FaceID again because I can't test it on iOS.

Also fixed an error when pressing back while on POS screen in certain situations.

Additionally removed some unnecessary `await`s.

This pull request is categorized as a:

- [ ] New feature
- [x] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance 
- [ ] Other

## Checklist
- [x] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [x] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [x] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [ ] No, I’m a fool
- [ ] Yes
- [x] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [x] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [x] LND (REST)
- [ ] LND (Lightning Node Connect)
- [ ] Core Lightning (c-lightning-REST)
- [ ] Core Lightning (Spark)
- [ ] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://app.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `yarn` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: views/Wallet/Wallet.tsx [hunk 1]
```diff
@@ -350,7 +353,7 @@ export default class Wallet extends React.Component<WalletProps, WalletState> {
         const error = NodeInfoStore.error || SettingsStore.error;
         const { implementation, settings, loggedIn, connecting, posStatus } =
             SettingsStore;
-        const loginRequired = SettingsStore.loginRequired();
+        const loginRequired = !settings || SettingsStore.loginRequired();
```
Comment (kaloudis) [inline views/Wallet/Wallet.tsx:356]: why require login if no settings here?

Comment (myxmaster) [reply views/Wallet/Wallet.tsx:356]: This prevents rendering potentially sensitive information when settings are not loaded yet and we don't know if login is required.

Comment (kaloudis) [reply views/Wallet/Wallet.tsx:356]: have you tested the behavior on a fresh install here? feel like this would break something

Comment (myxmaster) [reply views/Wallet/Wallet.tsx:356]: Just checked it by clearing storage and running Zeus again, configured the node and it worked.

**Pull request merged**